### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by ntjohns10**
basic authorization and player complete.  If there is not token, user is brought to login page, upon successful login user is brought to player page which should automatically begin playing  song.

Make sure your .env file has the spotify client_id and spotify user_secret.

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -9,8 +10,9 @@ export default function App() {
   const providerSongValue = useMemo(() => ({ songs, setSongs }), [songs, setSongs]);
 
   return (
-    <SongContext.Provider value={providerSongValue}>
-      <SearchBar />
-    </SongContext.Provider>
+    <div>
+      Hello Sonar!
+      <Landing/>
+    </div>
```
Comment (Panlord) [inline client/src/components/App.jsx:16]: Regarding these changes, wouldn't you want to resolve this conflict with something like
```
<SongContext...>
   <SearchBar />
  <Landing />
</SongContext>
```
I think with this here, you're deleting the SearchBar component that others wrote. That is, unless I am misinterpreting the code and/or this is intended.

File: client/src/components/landing.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      const json = await response.json();
+      setToken(json.access_token);
+    }
+
+    getToken();
+  }, []);
+
+  return (
+    <>
+      { (token === '') ? <Login/> : <Player token={token} />}
```
Comment (Panlord) [inline client/src/components/landing.jsx:21]: Your conditional rendering of this using a ternary  operator is clever and looks nice here. I commend you on this.

**Review by Panlord — CHANGES_REQUESTED** [review]
Code looks good overall, but I've left several comments. Once they have been addressed, I will be happy to approve of these changes.

File: client/src/components/App.jsx (hunk 1 above)
Comment (Panlord) [reply client/src/components/App.jsx:16]: Turns out that this was indeed an issue with my own interpretation. Indeed we do not want the search bar in the landing page. I would just let the SearchBar team know that this SearchBar component will need to be relocated elsewhere.

**Review by Panlord — APPROVED** [review]
Approved

**Review by toanddao — APPROVED** [review]
This looks great

**Pull request merged**